### ISSUE TITLE
Ambiguous error: thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: InvalidSep

### ISSUE THREAD

**Issue by qtoussaint** [rderelle/fastlin#2]
I'm running fastlin using the following command:

`fastlin -d path/to/sequence_reads -b MTBC_barcodes_for_fastlin.tsv -x 80`

and encountered the following error partway through the run:
```
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: InvalidSep { found: 53, pos: ErrorPosition { line: 3186683, id: Some("ERR3287584.796671") } }', src/analyse_sample.rs:34:43
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```
After some investigation I determined that there was an issue with the read files of sample ERR3287584 (accessible at https://www.ebi.ac.uk/ena/browser/view/ERR3287584), apparently due to an issue during download. After removing the offending reads and redownloading from ENA, the error was resolved upon rerunning.

For reference, the damaged part of the file looked like:

> ...
> +
> AAFFFJJJJFFJFJFJJJJJJJJJJJJJJJJJJJJJFAJJFJJJJJJJJJJJFFJJJJJJJJJJJJFJJFJJJJJJAJFJAJFJJJJJJJJJJJJJJJFJJJJJJJJJJJJJJJJJJJJJJJJJJFJJJJJJJJJFJJJFJFJJJJJFFJ
> @ERR3287584.796670 K00198:305:HTTTMBBXX:2:1220:23855:39664/1
> GAGGTGGCCTGCCCGGACGGCAGGTAGTGCGCTTGGAAGTTGTTGACCCGCACACAGATCGGGTGCAACGACGTGCCGTCGACGGTGTTGCCGGCGCGAAACGAGTCGAACGCGGCCGGCGACGCCGAACAAAAACCGGGTCCGCCGTCG
> +
> A<AFFJFJJA77FJFJAF7AFFAF--A<AF<FF-7FJJAJJFJJJJJFJFJJJJJ-FJ-JJF-7-FFJJJJJJJJA<7JJJFJJF7-FJ<<JJAA<-FA<FAFAJ7AJJFJJJA<J7AFJJAF7FFFFFJJ<AJFJJJJ7-<AJFAFFFJ
> @ERR3287584.796671 K00198:305:HTTTMBBXX:2:1220:25824796663 K001CJJJJJJJJJGAGGGGGGGGGGGCGCCGTACCGCCCGGATCCGGCCTACCGCC7584.79CGCAACGGTGTTGCTTTGAGGA7TGGTGTATGGA
> +
> 5:HT:HTTTMBBXFFJJJJ<JJJJFJJJJJFJJJJJ
> @ERR3287JJJJJ
> @ERR32875FJFJJJJ7JJJJJ
> @ERR32875FJFJJJJ7JJJTGGTCGC05:HTTTMBBA-F-F<J7GGCCATCA87584.7965JJJJ7FFFFFJFFJJJFAJAAJJJJJCAACTACGCCG K00198:305:HTTTMBBXX:2:1220:243JJ-JA7<JJJJFJA<JJ6TJJJFFJJJJJFJ1JJJJJJJJJJGATG-77---7AJJJJJJJCGGATCCAGACGACGACAAACTGCT5JJJ-FJJAAAACCG8X:2:122FJJJJJJJ
> ...

May I suggest that a more informative error could be thrown in this case, perhaps using `.expect()` instead of unwrapping?

**Comment by rderelle** [comment]
Thanks a lot Jacqueline!

I prepared a new version of fastlin that handles this type of error (and other errors such as corrupted files). The idea will be to stop scanning the file if an error is detected and to report the error in the output.

Here is an example of output with 1 valid fastq file and 3 error files (dummy1 has a faulty record, dummy2 a pseudo gzip file and dummy3 is a corrupted gzip file):

> #sample	nb_files	k_cov	mixture	lineages	log_barcodes	log_errors
> SRR7496481	single	10	no	1.2.1.2.1 (11)	1 (18, 8, 10, 5, 10, 9, 6, 5, 11, 5), 1.2.1 (12, 16, 6, 8, 17, 19, 13, 9, 16, 7), 1.2.1.2 (15, 10, 6, 6, 8, 12, 5, 7, 15, 6), 1.2.1.2.1 (4, 8, 18, 7, 11, 8, 13, 11, 13)	
> dummy1	single	0	no			Error in file "reads/dummy1.fastq.gz": FASTQ parse error: sequence length is 150, but quality length is 50 (record 'ERR551806.5' at line 17).
> dummy2	single	0	no			Error in file "reads/dummy2.fastq.gz": invalid gzip header
> dummy3	single	0	no			Error in file "reads/dummy3.fastq.gz": corrupt deflate stream
> 

This way, fastlin won't crash in case of an error, which will be reported in the output file, and will move to the next sample.
Would that be ok?

Romain

ps: SKA2 also seems to crash with each of these 3 types of errors (tested with v0.3.4). Although in these cases the error messages are correctly displayed by SKA2.

**Comment by rderelle** [comment]
@qtoussaint 

Now available as v0.2.2.1 (not yet on Bioconda).

Thanks again
Romain
